src/bio2bel/sources/rhea.py

```diff
@@ -0,0 +1,121 @@
+        q = """
+            SELECT ?compound_name ?chebi ?reactivePart WHERE {
+                ?side rh:contains ?participant .
+                ?participant rh:compound ?compound .
+                ?compound rh:name ?compound_name .
+                OPTIONAL {
+                    ?compound rh:reactivePart ?part_id .
+                    ?part_id rh:chebi ?chebi .
+                    ?part_id rh:name ?reactivePart .
+                } .
+                OPTIONAL {?compound rh:chebi ?chebi} .
+                OPTIONAL {?compound rh:underlyingChebi ?chebi}
+            }
+            """
+        result = g.query(q, initBindings={'side': side_uri})
+        # Get an iterable of the compounds (no need to remove duplicates since dictionary keys must be unique)
+        compounds = map(lambda x: x[0], result)
+        # Create a dictionary that will contain the nodes associated with a compound (for GenericCompounds with multiple ReactiveParts)
+        # Goal: for compounds with multiple ReactiveParts, link those reactiveParts together so they can easily be handled later
+        #   For now, we will ignore these compounds.
+        nodes_by_compound: Dict[str, List[dsl.BaseEntity]] = {c: [] for c in compounds}
+        for r in set(result):
+            compound_name, chebi_uri, reactive_part_name = r
+            # Based on the way the OPTIONAL modifier works, the SELECT query may return ?compound_name entries by themselves, without any ?chebi information
+            if not chebi_uri:
+                continue
+            # Remove the namespace from the chebi_uri to access the identifier
+            identifier = chebi_uri.replace(CH_NAMESPACE, '')
+            # Use the compound name if no reactive part is available
+            name = reactive_part_name if reactive_part_name else compound_name
+            # Build an abundance node, and append it to the node list under the correct compound
+            node = dsl.Abundance(namespace=CHEBI, name=name, identifier=identifier)
+            nodes_by_compound[compound_name].append(node)
+
+        for compound, node_list in nodes_by_compound.items():
+            node = None
+            if len(node_list) == 1:
+                node = node_list[0]
+            else:
+                # If there are multiple nodes, there must be multiple reactiveParts
+                # We have no idea how to handle this for the moment... skip it!
+                logger.debug(
+                    f'Encountered compound {compound} with multiple ReactiveParts {node_list} ...skipped.\n\
+                    PyBEL currently has no easy way to represent this type of entity, since the groupings contain meaningful information like stoichiometry and catalysis.\n \
+                    See: https://www.rhea-db.org/rhea/24960.',
+                )
+                continue
+            # Append the node to the corresponding list in participants
+            participants[i].append(node)
```

Comment:
it might be less efficient, but using a list of dictionaries instead of two parallel lists could make this a bit more organized

Reply:
Oh, as in have `participants` be a dictionary like the following?
```
participants
     |---'reactants'
                |---[node1, node2, ...]
     |---'products'
                |--- ...
```

Reply:
I might not understand your diagram, but I think it could look like this:
```python
participants = [
    {'reactants': ..., 'products': ...},
    {'reactants': ..., 'products': ...},
    ...
]

Reply:
Yep, that looks good! The `participants` variable only tracks the reactants/products for a single reaction, though, so I'll just change it to a single dictionary with keys 'reactants' and 'products.'